app/components/Table/Filters/TableFilter.tsx

```diff
@@ -4,6 +4,8 @@ import React from "react";
 import type { FilterComponent } from "./types";
 
 export interface FilterSettings {
+  isDisableClearable?: boolean;
+  isFreeSoloEnabled?: boolean;
```

Comment:
These shouldn't live here, since they're specific to the SearchableDropdownFilter.
What we should do is:
- extend the `FilterSettings` interface to another interface that includes those extra config options
- make the constructor of `SearchableDropdownFilter` use that new interface as the settings object type.
The `EnumFilter` is a good example of doing that


Also, can we name these "hideClearButton" and "allowFreeFormInput" or something like that? I don't expect devs to know what those particular mui-autocomplete settings do by just looking at the name